cmd/up/deploy.go

```diff
@@ -0,0 +1,140 @@
+// Copyright 2025 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package up
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/okteto/okteto/cmd/deploy"
+	pipelineCMD "github.com/okteto/okteto/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/analytics"
+	"github.com/okteto/okteto/pkg/cmd/pipeline"
+	"github.com/okteto/okteto/pkg/env"
+	oktetoErrors "github.com/okteto/okteto/pkg/errors"
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/model"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/spf13/afero"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// oktetoAutoDeployEnvVar if set the application will be deployed while running okteto up
+	oktetoAutoDeployEnvVar = "OKTETO_AUTODEPLOY"
+)
+
+// devEnvDeployerManager deploys the dev environment
+type devEnvDeployerManager struct {
+	isDevEnvDeployed func(ctx context.Context, name, namespace string, c kubernetes.Interface) bool
+
+	k8sClientProvider okteto.K8sClientProvider
+	ioCtrl            *io.Controller
+	getDeployer       func(deployParams) (deployer, error)
+}
+
+type deployer interface {
+	Run(ctx context.Context, opts *deploy.Options) error
+	TrackDeploy(manifest *model.Manifest, runInRemoteFlag bool, startTime time.Time, err error)
+}
+
+type deployParams struct {
+	deployFlag                     bool
+	okCtx                          *okteto.Context
+	devenvName, ns                 string
+	manifestPathFlag, manifestPath string
+	manifest                       *model.Manifest
+}
+
+// NewDevEnvDeployerManager creates a new DevEnvDeployer
+func NewDevEnvDeployerManager(up *upContext, ioCtrl *io.Controller, k8sLogger *io.K8sLogger) *devEnvDeployerManager {
+	return &devEnvDeployerManager{
+		ioCtrl:            ioCtrl,
+		k8sClientProvider: up.K8sClientProvider,
+		isDevEnvDeployed:  pipeline.IsDeployed,
+		getDeployer: func(params deployParams) (deployer, error) {
```

Comment:
Nit: As this function definition is a bit long, you could define the function outside, and use just the name here. It would help to see better what is being initialized. Anyways, non blocking